core/src/HIP/Kokkos_HIP_Space.hpp

```diff
@@ -65,6 +65,18 @@ class HIPSpace {
   ~HIPSpace()                              = default;
 
   /**\brief  Allocate untracked memory in the hip space */
+#ifdef KOKKOS_ENABLE_IMPL_HIP_UNIFIED_MEMORY
+  template <typename ExecutionSpace>
+  void* allocate(const ExecutionSpace&, const size_t arg_alloc_size) const {
+    return allocate(arg_alloc_size);
+  }
+  template <typename ExecutionSpace>
+  void* allocate(const ExecutionSpace&, const char* arg_label,
+                 const size_t arg_alloc_size,
+                 const size_t arg_logical_size = 0) const {
+    return allocate(arg_label, arg_alloc_size, arg_logical_size);
+  }
+#else
```

Comment:
```suggestion
#endif
```
Going forward, it still makes sense to treat the overload with a `HIP` execution space instance special.

Reply:
how so? It does the exact same thing as the one below?

Reply:
Yes, at the moment, but we will need a separate overload when using `hipMallocAsync` later.

Reply:
Then lets introduce it when we need it.